### PR TITLE
ROGER-946:Verify (and create if needed) test cases and functional tests for proxied auth

### PR DESCRIPTION
**Pull request by 04msambit**
**Added Following Tests:**

1 - Test user as "common_access" to /marathon/ui/* , /marathon/public/api/* etc
2 - Test PUT / GET / DELETE calls for a sample "internal_test_user" to following marathon end points:
      /marathon/v2/(apps|groups)/internal-test-team/
      /marathon/v2/(apps|groups)/+(sortdb|redis|mongo|ssdb)/internal-test-team/

3 - Test if an user is not able to do GET / PUT / DELETE calls to an unauthorized endpoint.
4 - Check if invalid end points throws error for REST calls. 

**Environment Tested On:**

- daldevmesoszk01
- daldevmesoszk02

### CODE REVIEW

**Comment by ankanm** [comment]
@amitbose327 @vmahedia - Can you please review?

File: test/nginx_aaad_proxy_test.py [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+    app_url = app_url = 'http://' + environment + ':4080/marathon/v2/tasks/'
+    result_code = get_data(app_url, headers)
+    assert(result_code == 200)
+    print("GET Passed for marathon/v2/tasks/\n")
+
+
+def test_proxy_user_valid_permissions(environment):
+
+    app_url = 'http://' + environment + ':4080/marathon/v2/apps/internal-test-team/test-app/'
+    encoded = base64.b64encode(b'internal_test_user:internal_test_user')
```
Comment (amitbose327) [inline test/nginx_aaad_proxy_test.py]: Can you move the base64 encoding part for the username:password into a method of its own. It is redundant code in all your test cases.

File: test/nginx_aaad_proxy_test.py [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+    assert(result_code == 200)
+    print("GET Passed for marathon/public/api")
+
+    app_url = app_url = 'http://' + environment + ':4080/marathon/v2/tasks/'
+    result_code = get_data(app_url, headers)
+    assert(result_code == 200)
+    print("GET Passed for marathon/v2/tasks/\n")
+
+
+def test_proxy_user_valid_permissions(environment):
```
Comment (amitbose327) [inline test/nginx_aaad_proxy_test.py]: This method seems monolithic and testing too many stuff. If it fails debugging becomes tougher. Can you break it into different test cases, maybe one for each HTTP method. Also separate our the multiple endpoints ( app_urls) into separate test cases.

File: test/nginx_aaad_proxy_test.py (hunk 1 above)
Comment (04msambit) [inline test/nginx_aaad_proxy_test.py]: Added separate functions to calculate headers.

File: test/nginx_aaad_proxy_test.py (hunk 2 above)
Comment (04msambit) [inline test/nginx_aaad_proxy_test.py]: Divided into smaller functions , to test  PUT/GET/DELETE  , for different end points.

File: test/nginx_aaad_proxy_test.py [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+    result = requests.put(url, data=json.dumps(payload), headers=headers)
+    return (result.status_code)
+
+
+def delete_data(url, headers, payload):
+    result = requests.delete(url, data=json.dumps(payload), headers=headers, auth=('internal_test_user', 'internal_test_user'))
+    return (result.status_code)
+
+
+def headers_value():
```
Comment (amitbose327) [inline test/nginx_aaad_proxy_test.py]: Can you rename this to get_authorization_header().

**Comment by amitbose327** [comment]
LGTM, apart from one minor comment.

File: test/nginx_aaad_proxy_test.py (hunk 3 above)
Comment (04msambit) [inline test/nginx_aaad_proxy_test.py]: Modified name to get_authorization_header()

**Pull request merged**